app/src/features/FhirResourceTree/resourceTreeSlice.ts

```diff
@@ -52,11 +63,67 @@ const resourceTreeSlice = createSlice({
         if (node) node.children = data.children;
       }
     },
+    setAttributeNodes: (
+      state,
+      { payload }: PayloadAction<{ attributeNodes: ElementNode[] }>
+    ) => {
+      const { attributeNodes } = payload;
+      const { attributeNodesInTree, root } = state;
+      if (root) {
+        const nodesToRemove = differenceBy(
+          attributeNodesInTree,
+          attributeNodes,
+          (node) => node.path
+        );
+        const nodesToAdd = differenceBy(
+          attributeNodes,
+          attributeNodesInTree,
+          (node) => node.path
+        );
+
+        // Add attribute nodes to the tree
+        nodesToAdd.forEach((node) => {
+          const parent = getNode(
+            "path",
+            computePathWithoutIndexes(node),
+            root as ElementNode
+          );
+          if (
+            parent &&
+            !parent.children.some(({ path }) => path === node.path)
+          ) {
+            parent.children.push(node);
+            attributeNodesInTree.push(node);
+          }
+        });
+
+        // Remove attribute nodes from the tree
+        nodesToRemove.forEach((node) => {
+          const parent = getNode(
+            "path",
+            computePathWithoutIndexes(node),
+            root as ElementNode
+          );
+          if (parent) {
+            parent.children = parent.children.filter(
+              ({ path }) => path !== node.path
+            );
+            state.attributeNodesInTree = attributeNodesInTree.filter(
+              ({ path }) => path !== node.path
+            );
+          }
+        });
+      }
+    },
   },
 });
 
 export const selectRoot = (state: RootState): ElementNode | undefined =>
   state.resourceTree.root;
 
-export const { setNodeChildren } = resourceTreeSlice.actions;
+export const {
+  setNodeChildren,
+  setAttributeNodes,
+  resetResourceTreeSliceState,
```

Comment:
maybe we could find better names for those actions https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters